lib/Dialect/DC/DCOps.cpp

```diff
@@ -0,0 +1,240 @@
+//===- DCOps.cpp ----------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/DC/DCOps.h"
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/FunctionImplementation.h"
+#include "mlir/IR/OpImplementation.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Interfaces/SideEffectInterfaces.h"
+
+using namespace circt;
+using namespace dc;
+using namespace mlir;
+
+bool circt::dc::isI1ValueType(Type t) {
+  auto vt = t.dyn_cast<ValueType>();
+  if (!vt || vt.getInnerTypes().size() != 1)
+    return false;
+
+  auto innerWidth = vt.getInnerTypes()[0].getIntOrFloatBitWidth();
+  return innerWidth == 1;
+}
+
+namespace circt {
+namespace dc {
+#include "circt/Dialect/DC/DCCanonicalization.h.inc"
+
+// =============================================================================
+// FuncOp
```

Comment:
I'm just going to ignore these under the assumption that they'll go away before a merge.